### ISSUE TITLE
Incorrect GetHttpMessageHandler logic in .NET6

### ISSUE THREAD

**Issue by marek-safar** [xamarin/xamarin-android#5753]
The current implementation of GetHttpMessageHandler is [incorrect](https://github.com/xamarin/xamarin-android/blob/main/src/Mono.Android/Android.Runtime/AndroidEnvironment.cs#L344). It returns by default null value which is not supported in .NET6.

@steveisok @pjcollins @jonathanpeppers

**Comment by jonpryor** [comment]
@steveisok: wouldn't it be better to instead update this line:

https://github.com/dotnet/runtime/blob/03be59b0cb4ee4914dc4c195bc7fd839d7b46d44/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.Android.cs#L26

```csharp
return (HttpMessageHandler)handlerMethod!.Invoke(null, null)!;
```

to instead do:

```csharp
if (handlerMethod != null) {
    return (HttpMessageHandler)handlerMethod!.Invoke(null, null)!;
}
return new HttpClientHandler();
```

**Comment by marek-safar** [comment]
> wouldn't it be better

Wouldn't that be odd that if someone explicitly says to use native handler it'd create SocketHttpHandler?

**Comment by grendello** [comment]
This was fixed in 1e5bfa3ff0ae18ce9407677667f05a9bfba53478
